pandas/plotting/_matplotlib/core.py

```diff
@@ -220,6 +224,98 @@ def __init__(
 
         self._validate_color_args()
 
+    def _validate_subplots_kwarg(
+        self, subplots: bool | Sequence[Sequence[str]]
+    ) -> bool | list[tuple[int, ...]]:
+        """
+        Validate the subplots parameter
+
+        - check type and content
+        - check for duplicate columns
+        - check for invalid column names
+        - convert column names into indices
+        - add missing columns in a group of their own
+        See comments in code below for more details.
+
+        Parameters
+        ----------
+        subplots : subplots parameters as passed to PlotAccessor
+
+        Returns
+        -------
+        validated subplots : a bool or a list of tuples of column indices. Columns
+        in the same tuple will be grouped together in the resulting plot.
+        """
+
+        if isinstance(subplots, bool):
+            return subplots
+        elif not isinstance(subplots, Iterable):
+            raise ValueError("subplots should be a bool or an iterable")
+
+        supported_kinds = (
+            "line",
+            "bar",
+            "barh",
+            "hist",
+            "kde",
+            "density",
+            "area",
+            "pie",
+        )
+        if self._kind not in supported_kinds:
+            raise ValueError(
+                "When subplots is an iterable, kind must be "
+                f"one of {', '.join(supported_kinds)}. Got {self._kind}."
+            )
+
+        if any(
+            not isinstance(group, Iterable) or isinstance(group, str)
+            for group in subplots
+        ):
+            raise ValueError(
+                "When subplots is an iterable, each entry "
+                "should be a list/tuple of column names."
+            )
+
+        cols_in_groups = [col for group in subplots for col in group]
+        duplicates = {col for (col, cnt) in Counter(cols_in_groups).items() if cnt > 1}
+        if duplicates:
+            raise ValueError(
+                f"Each column should be in only one subplot. Columns {duplicates} "
```

Comment:
Is it really a desirable behavior?
Why not allowing one to plot the same line against different sets?

Reply:
Currently (in master), only one column per plot is possible.
Allowing more than one column per plot should be possible, but would make the PR more complex. This PR il already 1.5+ years long so I would leave that improvement for a subsequent PR.

Reply:
yeah, agree, could be a follow-up enhancement PR!